account_prefs_update.php

```diff
@@ -84,6 +84,9 @@
 	$t_prefs->language = $t_lang;
 }
 
+$t_font = gpc_get_string( 'font_family' );
+config_set ( 'font_family', $t_font, $f_user_id );
```

Comment:
I would only call config_set() if it is different from the current effective value.  This way we don't save it unless necessary and hence later on changing the font family affects all users that didn't explicitly change it and just saved user preferences before.

Reply:
👍 
 also to keep the number of rows small in config table, especially as we always read the **whole** config table (see function config_cache_all)  

A bit off topic at this place, but even with this optimization, we could end in severe performance and memory issues in long term, at least on installations with lots of users.

I am aware that there are already other user preference options stored in this table, e.g. *_columns and email_notifications_verbose
This might have been a wrong decision and was just to avoid schema changes in user_pref table

I know, this is just one small new setting, but our next similar setting will also be just one small new setting.

We could create a new table for user preferences similar to the config table, where we fetch just data filtered by the current user id.
After that, migrate settings from user_pref and config table, finally drop user_pref table.